compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIEDistributeCoresAndObjectFifos.cpp

```diff
@@ -659,7 +648,10 @@ LogicalResult insertLogicalObjectFifoAccess(ModuleOp moduleOp) {
         opToInsertRewriterPoint = opToInsertRewriterPoint->getParentOp();
       }
       for (auto &&[idx, operand] : llvm::enumerate(op->getOpOperands())) {
-        if (memrefToLogicalObjectFifoAccess.contains(operand.get())) {
+        Operation *operandDefiningOp = operand.get().getDefiningOp();
+        if (!operandDefiningOp || !isa<memref::AllocOp>(operandDefiningOp)) {
+          // We're only transforming accesses to AllocOps here.
```

Comment:
I think you can just use `if(!isa_and_present<memref::AllocOp>(operand.getDefiningOp())`? And under if condition, it's not doing anything?

Reply:
> And under if condition, it's not doing anything?
It's preventing it from entering a later else if.

Reply:
Okay. I'll let @Abhishek-Varma to take a look as well.

Reply:
That's correct, we should use `dyn_cast_if_present` - it'd serve both the purpose.

I've added that as a suggestion in a separate thread along with a way to not go with empty bodied `if`.